Project-Hotpot/Project-Hotpot/Managers/CacheManager.swift

```diff
@@ -103,7 +103,7 @@ class CacheManager: NSObject {
                         //throws error later on
                         return nil
                     }
-                    return SongDetails(uri: uri, id: id, danceability: danceability, energy: energy, tempo: tempo, key: key)
+                    return SongDetails(uri: uri, id: id, danceability: Float(danceability), energy: Float(energy), tempo: Float(tempo), key: Float(key))
```

Comment:
Float causes some weird math and rounding issues, maybe use double (https://www.codingem.com/double-vs-float-swift/)